### PR TITLE
Fix crash in DefiniteAssignmentPass.VisitLocal

### PR DESCRIPTION
**Pull request by AlekseyTs**
Fixes https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1243877.

### CODE REVIEW

**Review by cston — APPROVED** [review]

File: src/Compilers/CSharp/Portable/FlowAnalysis/DefiniteAssignment.cs [hunk 1]
```diff
@@ -1162,7 +1162,12 @@ private bool IsAssigned(BoundExpression node, out int unassignedSlot)
             }
 
             Debug.Assert(unassignedSlot > 0);
-            return this.State.IsAssigned(unassignedSlot);
+            if (unassignedSlot > 0)
+            {
```
Comment (Youssef1313) [inline src/Compilers/CSharp/Portable/FlowAnalysis/DefiniteAssignment.cs:1166]: If the if condition can be met, then the assertion can fail. Should the assertion be removed?

Comment (AlekseyTs) [reply src/Compilers/CSharp/Portable/FlowAnalysis/DefiniteAssignment.cs:1166]: > If the if condition can be met, then the assertion can fail. Should the assertion be removed?

We are still expecting the assert to not fail, even though we don't to crash if it is.

---
In reply to: [537702704](https://github.com/dotnet/roslyn/pull/49828#discussion_r537702704) [](ancestors = 537702704)

**Comment by AlekseyTs** [comment]
@dotnet/roslyn-compiler Please review, need a second sign-off for a small change fixing a Watson crash.

**Review by RikkiGibson — APPROVED** [review]
I'm wondering if there are any other places where checks should be inserted after a call to GetOrCreateSlot, but there are a lot of calls to that and many indirect consumers, so I am not sure.

**Comment by AlekseyTs** [comment]
> I'm wondering if there are any other places where checks should be inserted after a call to GetOrCreateSlot, but there are a lot of calls to that and many indirect consumers, so I am not sure.

I reviewed all direct consumers of this API before sending this PR. 

---
In reply to: [547490465](https://github.com/dotnet/roslyn/pull/49828#issuecomment-547490465) [](ancestors = 547490465)

**Pull request merged**